jwst/pipeline/calwebb_image3.py

```diff
@@ -41,77 +40,63 @@ class Image3Pipeline(Pipeline):
-                # Set up Level 2c suffix to be used later
-                asn_id = input_models.meta.asn_table.asn_id
-                suffix_2c = '{}_{}'.format(asn_id, 'crf')
-                for model in input_models:
-                    self.save_model(
-                        model,
-                        output_file=model.meta.filename,
-                        suffix=suffix_2c
-                    )
-
-        self.log.info("Resampling images to final result...")
-        result = self.resample(input_models)
-
-        try:
-            result.meta.asn.pool_name = input_models.meta.asn_table.asn_pool
-            result.meta.asn.table_name = os.path.basename(input)
-            result.meta.filename = input_models.meta.asn_table.products[0].name
-        except Exception:
-            pass
-
-        self.save_model(result, suffix=self.suffix)
-
-        self.log.info("Creating source catalog...")
+
+        # Configure settings for saving results files
+        self.outlier_detection.suffix = 'crf'
+        self.outlier_detection.save_results = self.save_results
+
+        self.resample.suffix = 'i2d'
+        self.resample.save_results = self.save_results
+
         self.source_catalog.save_results = self.save_results
-        self.source_catalog(result)
-        # NOTE: source_catalog step writes out the catalog in .ecsv format
-        # In the future it would be nice if it was returned to the pipeline,
-        # and then written here.  A datamodel for .ecsv might be required.
 
-        return
+        with datamodels.open(input_data, asn_exptypes=asn_exptypes) as input_models:
+            # If input is an association, set the output to the product name.
+            try:
+                self.output_file = input_models.meta.asn_table.products[0].name
+            except AttributeError:
+                pass
+
+            # Check if input is single or multiple exposures
+            try:
+                has_groups = len(input_models.group_names) > 1
+            except Exception:
+                has_groups = False
+
+            if isinstance(input_models, datamodels.ModelContainer) and has_groups:
+                if is_moving_target(input_models):
+                    input_models = self.assign_mtwcs(input_models)
+                else:
+                    input_models = self.tweakreg(input_models)
+
+                input_models = self.skymatch(input_models)
+                input_models = self.outlier_detection(input_models)
+
+            result = self.resample(input_models)
+
+            try:
+                result.meta.asn.pool_name = input_models.meta.asn_table.asn_pool
+                result.meta.asn.table_name = os.path.basename(input_data)
+                result.meta.filename = input_models.meta.asn_table.products[0].name
+            except Exception:
+                pass
+
+            self.source_catalog(result)
+            # NOTE: source_catalog step writes out the catalog in .ecsv format
```

Comment:
This comment seemed to be there originally, but is nonsensical. This is operating by design. IMHO this comment should be removed.

Reply:
I think this comment was written based on 1) the assumption that the source catalog product was the final, main product of this pipeline, and hence the creation of the file should be handled by the pipeline, and 2) the pipeline was not able to save that file, because there's no corresponding data model for it (hence it must be saved by the step itself).

But in reality it's the resampled (i2d) product that is regarded as the final, main product and that's the one that's saved by the pipeline (or by stpipe when the pipeline finishes). Hence you're correct that this comment really doesn't apply and can be removed.